test/dist_autograd_test.py

```diff
@@ -433,10 +501,24 @@ def test_no_graph_with_tensors_not_require_grad(self):
 
             # Wait for the prev rank to be done with rpc.
             self._check_rpc_done(1)
-            # prev context id is not passed over as tensors do not require grads
-            with self.assertRaises(RuntimeError):
-                ctx = dist_autograd._retrieve_context(ctx_ids[1])
+            if ExecMode.RPC_SYNC == exec_mode:
+                # prev context id is not passed over as tensors do not require
+                # grads
+                with self.assertRaises(RuntimeError):
+                    ctx = dist_autograd._retrieve_context(ctx_ids[1])
+            elif ExecMode.REMOTE == exec_mode:
+                # NB: RRef.to_here() always passes the autograd context to the
+                # the callee, as the caller does not know whether the return
+                # value would contain a requires_grad tensor or not.
+                pass
+
+    @dist_init(setup_model_parallel=True)
+    def test_no_graph_with_tensors_not_require_grad(self):
+        self._test_no_graph_with_tensors_not_require_grad(ExecMode.RPC_SYNC)
 
+    @dist_init(setup_model_parallel=True)
+    def test_no_graph_with_tensors_not_require_grad_remote(self):
```

Comment:
Can we test nested RPCs with RRef? For example A shares RRef to B, B shares it to C, C calls to_here() does some computation and finally the result is retrieved on A and A calls backward on this. (The RRef could live on D).